response_operations_ui/controllers/cir_controller.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+
+import requests
+from flask import current_app as app
+from google.auth.exceptions import GoogleAuthError
+from structlog import wrap_logger
+
+from response_operations_ui.common.credentials import fetch_and_apply_oidc_credentials
+from response_operations_ui.exceptions.error_codes import (
+    ErrorCode,
+    get_error_code_message,
+)
+from response_operations_ui.exceptions.exceptions import ExternalApiError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+TARGET_SERVICE = "cir"
+
+
+def get_cir_service_status():
+    return _get_response_content(app.config["CIR_API_URL"] + "/status")
+
+
+def _get_response_content(request_url):
+    session = requests.Session()
+    client_id = app.config["CIR_OAUTH2_CLIENT_ID"]
+    logger.debug(f"{TARGET_SERVICE} service request", session_headers=str(session.headers), request_url=request_url)
+
+    try:
+        fetch_and_apply_oidc_credentials(session=session, client_id=client_id)
```

Comment:
If we had other use cases in this application for other OIDC/IAP clients, we might want to use a decorator to wrap this oidc_credentials stuff. But given cir is the only client use case at the moment, visible and explicit here is fine.